pkg/base1/cockpit.js

```diff
@@ -153,30 +153,54 @@ function transport_debug() {
 }
 
 function event_mixin(obj, handlers) {
-    obj.addEventListener = function addEventListener(type, handler) {
-        if (handlers[type] === undefined)
-            handlers[type] = [ ];
-        handlers[type].push(handler);
-    };
-    obj.removeEventListener = function removeEventListener(type, handler) {
-        var length = handlers[type] ? handlers[type].length : 0;
-        for (var i = 0; i < length; i++) {
-            if (handlers[type][i] === handler) {
-                handlers[type][i] = null;
-                break;
+    Object.defineProperties(obj, {
+        addEventListener: {
+            enumerable: false,
+            value: function addEventListener(type, handler) {
+                if (handlers[type] === undefined)
+                    handlers[type] = [ ];
+                handlers[type].push(handler);
+            }
+        },
+        removeEventListener: {
+            enumerable: false,
+            value: function removeEventListener(type, handler) {
+                var length = handlers[type] ? handlers[type].length : 0;
```

Comment:
> We're not worried about concurrency but reentrancy. The code above will break the latter.

Why? Where do we store an index to an entry in the array for later use? Other than that I don't see how the javascript execution model would allow problems to arise from shrinking the array.